devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,392 @@
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name: "temperature", type: "generic", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.temperature", key: "PRIMARY_CONTROL") {
+				attributeState "temperature", label: '${currentValue}°',
+					backgroundColors: [
+						[value: 31, color: "#153591"],
+						[value: 44, color: "#1e9cbb"],
+						[value: 59, color: "#90d2a7"],
+						[value: 74, color: "#44b621"],
+						[value: 84, color: "#f1d801"],
+						[value: 95, color: "#d04e00"],
+						[value: 96, color: "#bc2323"]
+					]
+			}
+		}
+		valueTile("humidity", "device.humidity", inactiveLabel: false, width: 2, height: 2) {
+			state "humidity", label: '${currentValue}% humidity', unit: ""
+		}
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		main "temperature", "humidity"
+		details(["temperature", "humidity", "refresh"])
+	}
+}
+
+def installed() {
+	log.info "Installed called '${device.displayName}'"
+	configure()
+}
+
+def updated() {
+	log.info "updated called"
+	configure()
+}
+
+def configure() {
+	log.info "configure called"
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	def msrdata = getDataValue("MSR")
+	def commands = []
+	commands << zwave.multiChannelV3.multiChannelEndPointGet()
+	log.debug "configure: commands '${commands}'"
+	log.debug "configure: msrdata '${msrdata}'"
+	log.debug "configure: tempOffset '${tempOffset}'"
+	log.debug "configure: humidityOffset '${humidityOffset}'"
+
+	if (msrdata == null) {
+		commands << zwave.manufacturerSpecificV2.manufacturerSpecificGet()
+	}
+
+	if (tempOffset != null) {
+		commands << zwave.configurationV1.configurationSet(parameterNumber: 1, size: 1, scaledConfigurationValue: tempOffset as int)
+	}
+	if (humidityOffset != null) {
+		commands << zwave.configurationV1.configurationSet(parameterNumber: 2, size: 1, scaledConfigurationValue: humidityOffset as int)
+	}
+	commands << zwave.configurationV1.configurationGet(parameterNumber: 1)
+	commands << zwave.configurationV1.configurationGet(parameterNumber: 2)
+
+	commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x01) // temperature
+	commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 0x05) // humidity
+
+	sendCommands(commands,1000)
+}
+
+/**
+ * Mapping of command classes and associated versions used for this DTH
+ */
+private getCommandClassVersions() {
+	[
+		0x20: 1,  // Basic
+		0x25: 1,  // Switch Binary
+		0x30: 1,  // Sensor Binary
+		0x31: 5,  // Sensor MultiLevel
+		0x32: 3,  // Meter
+		0x56: 1,  // Crc16Encap
+		0x60: 3,  // Multi-Channel
+		0x70: 2,  // Configuration
+		0x84: 1,  // WakeUp
+		0x98: 1,  // Security
+		0x9C: 1,  // Sensor Alarm
+		0x71: 3   // Notification
```

Comment:
please remove command classes that are not being used in this DTH

Reply:
Ok. I will patch.